src/utils/builder.ts

```diff
@@ -69,46 +69,131 @@ export function setCriteriaElm(pathway: Pathway, elm: object): void {
   pathway.elm.criteria = elm;
 }
 
-function addState(pathway: Pathway, key?: string): string {
+export function createState(key?: string): State {
```

Comment:
Why does this need to be exported if the branch state and guidance state methods are exported?

Reply:
We need to export this to be used in the `BranchNode` component.

Reply:
Why can't it just use the `addBranchState` function?

Reply:
This was reworked to simplify the function. It will return a basic state object, which is then added to a pathway in `addNode()` in the sidebar component. This then lets us do things like add transitions and set the node type.